### PR TITLE
Implement circuit for CALLER and CALLVALUE opcode

### PR DESCRIPTION
**Pull request by ed255**
This PR depends on 
- [x] https://github.com/appliedzkp/zkevm-circuits/pull/279
- [x] https://github.com/appliedzkp/zkevm-circuits/pull/292

Based on https://github.com/appliedzkp/zkevm-specs/pull/90 and https://github.com/appliedzkp/zkevm-specs/pull/96

Take a special look at the approach I've followed to verify the equivalence between caller address in the StackOp (rlc encoded) and the CallContext (raw value) .

### CODE REVIEW

**Comment by CPerezz** [comment]
This will be a lot easier to review once #279 and #278 get merged. As the diff is quite big. 

Should we wait for now on the review? And also, I've already reviewed it while debugging. So probably @han0110 or others might have a more unaltered view on that! :)

**Comment by ed255** [comment]
> This will be a lot easier to review once #279 and #278 get merged. As the diff is quite big.

Completely agree!

> Should we wait for now on the review? And also, I've already reviewed it while debugging. So probably @han0110 or others might have a more unaltered view on that! :)

Yes I'd suggest waiting until #279 and #278 is merged before reviewing this PR.  I'll remove the draft mode once it's ready for review.

File: zkevm-circuits/src/evm_circuit/execution/caller.rs
Comment (CPerezz) [inline zkevm-circuits/src/evm_circuit/execution/caller.rs:78]: Don't we have a constant for the byte-size of an address? I remember we had it not so long ago.

**Review by CPerezz — APPROVED** [review]
LGTM

Just the const nit for the ADDR_BYTE_LENGTH

File: zkevm-circuits/src/evm_circuit/execution/caller.rs
Comment (ed255) [reply zkevm-circuits/src/evm_circuit/execution/caller.rs]: Thanks for spotting this!  I've updated the PR to use `N_BYTES_ACCOUNT_ADDRESS`

**Pull request merged**
